server/pulp/server/controllers/repository.py

```diff
@@ -229,24 +229,25 @@ def create_repo(repo_id, display_name=None, description=None, notes=None, import
             raise
 
     # Add the distributors. Delete the repository if this fails.
-    distributor_manager = manager_factory.repo_distributor_manager()
-    for distributor in distributor_list:
-        if not isinstance(distributor, dict):
-            repo.delete()
-            raise pulp_exceptions.InvalidValue(['distributor_list'])
-        try:
-            # Validation will occur in distributor manager.
-            type_id = distributor.get('distributor_type_id')
-            plugin_config = distributor.get('distributor_config')
-            auto_publish = distributor.get('auto_publish', False)
-            distributor_id = distributor.get('distributor_id')
-            distributor_manager.add_distributor(repo_id, type_id, plugin_config, auto_publish,
-                                                distributor_id)
-        except Exception:
-            _logger.exception('Exception adding distributor to repo [%s]; the repo will be '
-                              'deleted' % repo_id)
-            repo.delete()
-            raise
+    if distributor_list:
+        distributor_manager = manager_factory.repo_distributor_manager()
+        for distributor in distributor_list:
+            if not isinstance(distributor, dict):
```

Comment:
It would be better to do this check above, before creating the repo. You're already doing some validation on this value at line 206, so consider putting this there.